lib/Serialization/Deserialization.cpp

```diff
@@ -3376,37 +3376,58 @@ ModuleFile::getDeclCheckedImpl(DeclID DID) {
   case decls_block::PREFIX_OPERATOR_DECL: {
     IdentifierID nameID;
     DeclContextID contextID;
+    TypeID protoID;
 
-    decls_block::PrefixOperatorLayout::readRecord(scratch, nameID,
-                                                  contextID);
+    decls_block::PrefixOperatorLayout::readRecord(scratch, nameID, contextID,
+                                                  protoID);
     auto DC = getDeclContext(contextID);
-    declOrOffset = createDecl<PrefixOperatorDecl>(DC, SourceLoc(),
-                                                  getIdentifier(nameID),
-                                                  SourceLoc());
+    TypeLoc protoTyLoc = TypeLoc::withoutLoc(getType(protoID));
+    Identifier protocolName;
+    if (auto protoTy = protoTyLoc.getType())
+      if (auto *protoDecl = protoTy->getNominalOrBoundGenericNominal())
+        protocolName = protoDecl->getName();
```

Comment:
I only skimmed your previous proposal, but this confuses me: an identifier isn't enough to uniquely identify a type, so why bother?